feat: Added playground and preview feature

yatharth1706 opened the pull request:
## Fixes Issue #2645 

closes #2645 

## Changes proposed

1. Added playground page
2. Added preview page

## Note to reviewers
Created a draft pull request. @eddiejaoude once review and let me know what you think.

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/4211"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

eddiejaoude commented on components/user/UserProfile.js:
```diff
@@ -11,6 +11,7 @@ import { abbreviateNumber } from "../../services/utils/abbreviateNumbers";
 export default function UserProfile({ BASE_URL, data }) {
   const [qrShow, setQrShow] = useState(false);
   const fallbackImageSize = 120;
+  console.log(data);
```
I think this debug code can be removed

eddiejaoude commented on components/PreviewModal.js:
```diff
@@ -0,0 +1,187 @@
+import Link from "../components/Link";
+import { IconContext } from "react-icons";
+import { FaRegComments } from "react-icons/fa";
+import requestIp from "request-ip";
```
Are all these imports used?

eddiejaoude commented on components/PreviewModal.js:
```diff
@@ -0,0 +1,187 @@
+//     return <SingleLayout>{page}</SingleLayout>;
+//   }
+//   return <MultiLayout>{page}</MultiLayout>;
+// };
```
I think commented out code can be removed?

eddiejaoude commented on pages/playground.js:
```diff
@@ -0,0 +1,144 @@
+  };
+
+  const handleReset = () => {
+    localStorage.removeItem("PREVIEW_PROFILE_DATA");
```
Is local storage still needed if used in a modal on the same page?

eddiejaoude commented on components/PreviewModal.js:
```diff
@@ -0,0 +1,187 @@
+  }, [isOpen]);
+
+  const handlePreviewData = () => {
+    let userPreviewData = localStorage.getItem("PREVIEW_PROFILE_DATA");
```
I think this could be passed into the component as a `data` object

eddiejaoude commented on components/PreviewModal.js:
```diff
@@ -0,0 +1,187 @@
+    let userPreviewData = localStorage.getItem("PREVIEW_PROFILE_DATA");
+    let parsedData = JSON.parse(userPreviewData);
+    setUserData(parsedData);
+    console.log(parsedData);
```
I think the debug code can be removed?

yatharth1706 replied:
Thanks @eddiejaoude for reviewing the code. Will make all these changes which you suggested

yatharth1706 commented:
Hy @eddiejaoude, I have done all the changes which you suggested
- Removed local storage code
- Removed debug code as well like console.log statements
- Removed unused libraries
- Removed commented code
- Passing data prop to preview instead of taking data from local storage

eddiejaoude commented:
Looks good. Just had a quick play and when navigating to Testimonial tab I get an error

<img width="1118" alt="Screenshot 2023-02-08 at 07 26 14" src="https://user-images.githubusercontent.com/624760/217397556-64a23c9f-e2ab-4e98-9d1c-ba89aecece0e.png">

yatharth1706 commented:
> Looks good. Just had a quick play and when navigating to Testimonial tab I get an error
> 
> <img alt="Screenshot 2023-02-08 at 07 26 14" width="1118" src="https://user-images.githubusercontent.com/624760/217397556-64a23c9f-e2ab-4e98-9d1c-ba89aecece0e.png">

Ok @eddiejaoude, I will take a look at this why it is happening. Will solve it by EOD or tomorrow

yatharth1706 commented:
Hy @eddiejaoude testimonials one can be little tricky, I think. 

I was checking your code and whenever user go to /[username].  API is called which gets all the json and testimonials from data folder and combine all testimonials like its title, description, date in the json.

For my playground, I am not making any api calls. I am using the json which user provides in the UI. And according to that UI showing the preview. And for testimonials, I think will have to ask more information in the json. As our playground is totally dependent on json. Its not going to testimonials folder to find full info of testimonials given by particular user

Need your ideas as well. I think for now our playground will not work properly with testimonials. Either we should change the format of json and provide full information of testimonials. Currently in testimonials only gitusername is written. If we write full info like title, description, date along with git usernames. Then i can make changes accordingly to show that in preview.

eddiejaoude commented:
Thank @yatharth1706 for investigating 👍 I have some ideas, and I think the best solution would be to break the profile  page into smaller reusable components then you can use the components you want for your feature - I agree the testimonials are not needed for this, keep this feature simple would be best

The challenge though is doing this refactor of the app is a lot of work and will cause conflicts - plus I think should be in a separate PR to this one - I have started doing it but still a long way to go.

So maybe for now this PR is blocked by another PR and needs to be parked

yatharth1706 commented:
Hy @eddiejaoude, I actually didnt understand properly. So the other PR which you mentioned, are you working on it ?

eddiejaoude commented:
> Hy @eddiejaoude, I actually didnt understand properly. So the other PR which you mentioned, are you working on it ?

Sorry if my message was not clear. Yes I have started breaking it up into reusable components. I will see if there is a short term solution we can use to get this PR merged sooner

yatharth1706 commented:
Ok @eddiejaoude

eddiejaoude commented:
A recently merged PR might help this PR, I will take a look https://github.com/EddieHubCommunity/LinkFree/pull/4997/files#diff-45322352e129989f2db77f2dffd9308e2420ba561efa94d938cadb79fb27c5e4
